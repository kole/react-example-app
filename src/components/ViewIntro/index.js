import React from "react";
import PropTypes from "prop-types";
import { connect } from 'react-redux';
import styled from "styled-components";

import H1 from "~/components/H1";
import H2 from "~/components/H2";
import Button from "~/components/Button";
import { fetchQuestions } from '~/actions/questions';

const IntroContainer = styled.div`
    display: flex;
    flex: 1;
    flex-direction: column;
    justify-content: center;
    padding: 4em;
    text-align: center;

    > * {
        display: flex;
        flex: 1;
        flex-direction: column;
        justify-content: center;
    }
`;

class Intro extends React.Component {
    handleClick() {
        const { history, dispatch } = this.props;

        // User has requested to begin quiz, get questions
        dispatch(fetchQuestions());

        // go to the first question URL
        history.push("/question/1");
    }

    render() {
        return (
            <IntroContainer>
                <H1 text="Welcome to the Trivia Challenge" />
                <H2 text="You will be presented with 10 True or False questions." />
                <H2 text="Can you score 100%?" />
                <Button
                    clickHandler={this.handleClick.bind(this)}
                    text="Begin"
                />
            </IntroContainer>
        );
    }
}

Intro.propTypes = {
    dispatch: PropTypes.func.isRequired,
    history: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
    questions: state.questions.items,
    fetching: state.questions.fetching,
    error: state.questions.error
});

export default connect(mapStateToProps)(Intro);
