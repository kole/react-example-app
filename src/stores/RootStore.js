import Questions from './questions';
import Answers from './answers';
import Results from './results';

export default class RootStoreClass {
    constructor() {
        this.QuestionsStore = new Questions(this);
        this.AnswersStore = new Answers(this);
        this.ResultsStore = new Results(this);
    }
}
