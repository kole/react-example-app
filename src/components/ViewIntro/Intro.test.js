/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import ViewIntro from './index.js';

import RootStore from '~/stores/RootStore';
const store = new RootStore();

const mock = { history: {}};

describe('ViewIntro', () => {
    test('Snapshot', () => {
        const component = renderer
            .create(<ViewIntro history={mock.history} rootStore={store} />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
