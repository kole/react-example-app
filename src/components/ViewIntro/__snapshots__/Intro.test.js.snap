// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewIntro Snapshot 1`] = `
<div
  className="sc-bxivhb fmvafS"
>
  <div
    className="sc-bdVaJa iSJzkt"
  >
    Welcome to the Trivia Challenge
  </div>
  <div
    className="sc-bwzfXH dSwnPu"
  >
    You will be presented with 10 True or False questions.
  </div>
  <div
    className="sc-bwzfXH dSwnPu"
  >
    Can you score 100%?
  </div>
  <div
    className="sc-htpNat gYNKdz"
  >
    <button
      onClick={[Function]}
    >
      Begin
    </button>
  </div>
</div>
`;
